### PR TITLE
graceful import

### PR DESCRIPTION
**Pull request by harvey-intel**
Check import xgboost and version?

### CODE REVIEW

**Comment by harvey-intel** [comment]
@taylorjacklespriggs , want to get your eyes on this PR to see if it makes sense to explicitly surface some of these ImportError.

File: sigopt/xgboost/compat.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+MIN_REQUIRED_XGBOOST_VERSION = "1.3.0"
+
+
+try:
+  import xgboost
+  from xgboost import Booster, DMatrix
+except ImportError as ie_xgb:
+  raise ImportError(
+    "xgboost needs to be installed in order to use sigopt.xgboost.run functionality. "
+    "Try running `pip install sigopt [xgboost]`."
```
Comment (taylorjacklespriggs) [inline sigopt/xgboost/compat.py:11]: I think you'll want `pip install 'sigopt[xgboost]'`. I think pip will try to install the package `"[xgboost]"`, which does not exist. Also, some shells treat `[]` as special characters so they need to be in quotes.

File: sigopt/xgboost/compat.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  if installed_xgb_ver[0] < required_xgb_ver[0]:
+    satisfies_xgb_req = False
+  elif installed_xgb_ver[1] < required_xgb_ver[1]:
+    satisfies_xgb_req = False
+
+  if not satisfies_xgb_req:
+    raise ImportError(
+      f"sigopt.xgboost.run is compatible with xgboost>={MIN_REQUIRED_XGBOOST_VERSION}. "
+      f"You have version {xgboost.__version__} installed."
+    ) from None
```
Comment (taylorjacklespriggs) [inline sigopt/xgboost/compat.py:31]: gross. I suggest adding `packaging` into `requirements.txt`

**Review by taylorjacklespriggs — COMMENTED** [review]
I agree with this approach

**Pull request merged**